### PR TITLE
Test: implemented basic model swtich and model CRUD tests. 

### PR DESCRIPTION
**Pull request by PaulCCCCCCH**
## Issue ticket number and link

Issue #186

## What are the changes

Test: implemented basic model swtich and model CRUD tests. 
Changed model upload to call `RResponse.abort` instead of returning `RResponse.fail() object`

## Comments

This only includes basic cases to make sure happy path works. Will add more tests for corner cases later.

## Checklist before requesting a review

- [x] Double check the diff between my branch and the target branch. Make sure there's nothing unexpected.
- [x] Tag relevant issues and PRs in the first section of this form.
- [x] All CI tests are passed.
- [x] No `package-lock.json` or `package.json` changes are pushed. If you do, add a new section `Dependency Changes` to the form stating what library changes are introduced and why.

### CODE REVIEW

File: back-end/apis/model.py [hunk 1]
```diff
@@ -25,7 +26,7 @@ def set_curr_model(model_name: str):
         RServer.get_model_wrapper().set_current_model(model_name)
         return RResponse.ok("Success")
     except Exception as e:
-        RResponse.abort(500, "Failed to switch model." + str(e))
+        RResponse.abort(500, f"Failed to switch to model {model_name}. Error: {str(e)}")
```
Comment (Leon-Leyang) [inline back-end/apis/model.py:30]:
```suggestion
        traceback.print_exc()
        RResponse.abort(500, f"Failed to switch to model {model_name}. Error: {str(e)}")
```

File: back-end/apis/model.py [hunk 2]
```diff
@@ -36,7 +37,7 @@ def delete_model(model_name: str):
     try:
         model = RServer.get_model_wrapper().delete_model_by_name(model_name)
     except Exception as e:
-        RResponse.abort(500, f"Failed to delete model {model_name}." + str(e))
+        RResponse.abort(500, f"Failed to delete model {model_name}. Error: {str(e)}")
```
Comment (Leon-Leyang) [inline back-end/apis/model.py:42]:
```suggestion
        traceback.print_exc()
        RResponse.abort(500, f"Failed to delete model {model_name}. Error: {str(e)}")
```

**Review by Leon-Leyang — APPROVED** [review]
Nice work!!

**Review by Leon-Leyang — APPROVED** [review]
